src/app/clusters/level-control/level-control.cpp

```diff
@@ -190,36 +191,37 @@ void emberAfLevelControlClusterServerTickCallback(EndpointId endpoint)
     state->elapsedTimeMs += state->eventDurationMs;
 
     // Read the attribute; print error message and return if it can't be read
-    status = Attributes::CurrentLevel::Get(endpoint, &currentLevel);
-    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    status = LevelControl::Attributes::CurrentLevel::Get(endpoint, currentLevel);
+
+    if (status != EMBER_ZCL_STATUS_SUCCESS || currentLevel.IsNull())
     {
         emberAfLevelControlClusterPrintln("ERR: reading current level %x", status);
         writeRemainingTime(endpoint, 0);
         return;
     }
 
-    emberAfLevelControlClusterPrint("Event: move from %d", currentLevel);
+    emberAfLevelControlClusterPrint("Event: move from %d", currentLevel.Value());
 
     // adjust by the proper amount, either up or down
     if (state->transitionTimeMs == 0)
     {
         // Immediate, not over a time interval.
-        currentLevel = state->moveToLevel;
+        currentLevel.Value() = state->moveToLevel;
     }
     else if (state->increasing)
     {
-        assert(currentLevel < state->maxLevel);
-        assert(currentLevel < state->moveToLevel);
-        currentLevel++;
+        assert(currentLevel.Value() < state->maxLevel);
+        assert(currentLevel.Value() < state->moveToLevel);
+        currentLevel.SetNonNull(currentLevel.Value()++);
```

Comment:
This does not look right: it won't actually change `currentLevel`, right?  You might want:
```suggestion
        currentLevel.SetNonNull(currentLevel.Value() + 1);
```

Reply:
done